cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSICKANSink.java

```diff
@@ -285,9 +311,23 @@ public void capRecords(NGSIBatch batch, long maxRecords) throws CygnusCappingErr
             String entityForNaming = event.getEntityForNaming(enableGrouping, enableNameMappings, enableEncoding);
 
             try {
-                String orgName = buildOrgName(service);
-                String pkgName = buildPkgName(service, servicePathForNaming);
-                String resName = buildResName(entityForNaming);
+            	String orgName="";
+                String pkgName="";
+                String resName="";
+            	 if(!pkgByEntity) {
+                 	orgName = buildOrgName(service);
+                 	pkgName = buildPkgName(service, servicePathForNaming);
+                 	resName = buildResName(entityForNaming);
+                 }
+                 else if(pkgByEntity) {
```

Comment:
it's not necessary to add second if

Reply:
Fixed in 6a8704233ea63853e822d9868a6f93b429ea60d7